frontend/src/form/FormWebSocketContainer.tsx

```diff
@@ -0,0 +1,195 @@
+                ... on Checkbox {
+                  label
+                  booleanValue: value
+                }
+                ... on Select {
+                  label
+                  value
+                  options {
+                    id
+                    label
+                  }
+                }
+                ... on Radio {
+                  label
+                  options {
+                    id
+                    label
+                    selected
+                  }
+                }
+                ... on List {
+                  label
+                  items {
+                    id
+                    label
+                    imageURL
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`.loc.source.body;
+const formEventOperationName = 'formEvent';
+
+/**
+ * Connect the Form component to the GraphQL API over Web Socket.
+ */
+export const FormWebSocketContainer = ({ formId }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { viewState, form, displayedFormId, subscribers, widgetSubscriptions, message } = state;
+
+  const { graphQLWebSocketClient } = useContext(GraphQLClient);
+  const { id } = useProject() as any;
+
+  /**
+   * Displays an other form if the selection indicates that we should display another properties view.
+   */
+  useEffect(() => {
+    if (displayedFormId !== formId) {
+      dispatch({ type: SWITCH_FORM__ACTION, formId });
+    }
+  }, [formId, displayedFormId]);
+
+  useEffect(() => {
+    if (viewState === LOADING__STATE) {
+      dispatch({ type: INITIALIZE__ACTION });
+    }
+  }, [viewState]);
+
+  useEffect(() => {
+    if (viewState !== READY__STATE) {
+      return () => {};
+    }
+
+    const operationId = graphQLWebSocketClient.generateOperationId();
+    const subscribe = () => {
+      graphQLWebSocketClient.on(operationId, (message) => {
+        switch (message.type) {
+          case 'connection_error':
+            dispatch({ type: HANDLE_CONNECTION_ERROR__ACTION, message });
+            break;
+          case 'error':
+            dispatch({ type: HANDLE_ERROR__ACTION, message });
+            break;
+          case 'data':
+            dispatch({ type: HANDLE_DATA__ACTION, message });
+            break;
+          case 'complete':
+            dispatch({ type: HANDLE_COMPLETE__ACTION, message });
+            break;
+          default:
+            break;
+        }
+      });
+      const variables = {
+        input: {
+          projectId: id,
+          formId,
+        },
+      };
+      graphQLWebSocketClient.start(operationId, formEventSubscription, variables, formEventOperationName);
+    };
+
+    const unsubscribe = (id) => {
+      graphQLWebSocketClient.remove(id);
+      graphQLWebSocketClient.stop(id);
+    };
+
+    subscribe();
+    return () => unsubscribe(operationId);
+  }, [id, formId, graphQLWebSocketClient, viewState]);
```

Comment:
This will need to be converted to Apollo. I add it to my list.

Reply:
More specifically it should not, it should be merged with the one from the properties that you already have converted. No need to have the same content twice, it's useless. It will come with some of the impact of https://github.com/eclipse-sirius/sirius-components/issues/159

It's a bit complicated to do it now without a lot of conflicts between this PR and the upcoming one.